### PR TITLE
refactor: Refactoring python code + IDE stack

### PR DESCRIPTION
**Pull request by leandrodamascena**
Refactored some parts of Python and added some items to the IDE.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: functions/typescript/modules/module2/index.complete.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ const recordHandler = async (
   record: DynamoDBRecord,
   lambdaContext: Context
 ): Promise<void> => {
-  if (lambdaContext.getRemainingTimeInMillis() > 1000) {
+  if (lambdaContext.getRemainingTimeInMillis() < 1000) {
```
Comment (leandrodamascena) [inline functions/typescript/modules/module2/index.complete.ts:33]: I fixed this, ok?

File: infra/lib/attendant-ide/compute-construct.ts [hunk 2]
```diff
@@ -54,8 +58,11 @@ export class ComputeConstruct extends Construct {
     userData.addCommands(
       // Add .NET repo
       `rpm -Uvh ${dotNetRepo}`,
+      `yum clean all`,
       // Install general dependencies
-      `yum install -y  ${osPackages.join(' ')}`,
+      `yum install -y --downloadonly  ${osPackages.join(' ')}`,
+      `sleep 20`,
+      `yum install -y ${osPackages.join(' ')}`,
```
Comment (leandrodamascena) [inline infra/lib/attendant-ide/compute-construct.ts:65]: I have no idea why, but `yum` was finishing the transaction before creating the cache and crashing randomly.

Using this solution we first download packages and after we can install them.


```sh
Total                                            46 MB/s | 359 MB     00:07
[Errno 2] No such file or directory: '/var/cache/dnf/amazonlinux-87b6c41b372540cd/packages/langpacks-core-font-en-3.0-21.amzn2023.0.4.noarch.rpm'
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'yum clean packages'.
Redirecting to /bin/systemctl start docker.service
Failed to start docker.service: Unit docker.service not found.
```

File: infra/lib/attendant-ide/compute-construct.ts [hunk 3]
```diff
@@ -129,7 +136,10 @@ export class ComputeConstruct extends Construct {
         `export PATH="$HOME/.local/share/fnm:$PATH"`,
         `eval "\`fnm env\`"`,
         `cd /home/${whoamiUser}/${workshopDirectory}`,
-        `npm ci`
+        `npm ci`,
+        `npm run utils:createConfig`,
+        `npm run frontend:build`,
+        `aws s3 cp --recursive ./frontend/build s3://${props.websiteBucketName}`
```
Comment (leandrodamascena) [inline infra/lib/attendant-ide/compute-construct.ts:142]: I need to fix this:

Trying to find output file locally.
Unable to find output file locally, trying remotely.
Trying to find stack with PowerToolsWorkshop
Error: Unable to find stack among loaded ones
    at getStackName (file:///home/ec2-user/workshop/scripts/create-aws-exports.mjs:41:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (file:///home/ec2-user/workshop/scripts/create-aws-exports.mjs:136:21)
Unable to load CloudFormation stacks.
Did you run `npm run infra:deploy` in the project root?
file:///home/ec2-user/workshop/scripts/create-aws-exports.mjs:143
      throw new Error('Unable to find parameters locally or remotely.');

File: infra/lib/ide-stack.ts [hunk 4]
```diff
@@ -16,9 +17,13 @@ export class IdeStack extends Stack {
     const network = new NetworkConstruct(this, 'network', {});
     const { vpc } = network;
 
+    // Import the WebsiteBucketName output from the source stack (SourceStack)
+    const websiteBucketName = Fn.importValue('PowerToolsWorkshop-WebsiteBucketName');
```
Comment (leandrodamascena) [inline infra/lib/ide-stack.ts:20]: Getting value from the exported value.

File: infra/lib/frontend/storage-construct.ts [hunk 5]
```diff
@@ -57,6 +57,7 @@ export class StorageConstruct extends Construct {
 
     new CfnOutput(this, 'WebsiteBucketName', {
       value: this.websiteBucket.bucketName,
+      exportName: `${Stack.of(this).stackName}-WebsiteBucketName`,
```
Comment (leandrodamascena) [inline infra/lib/frontend/storage-construct.ts:60]: Exporting bucket name

**Review by leandrodamascena — COMMENTED** [review]
Hey @dreamorosi! I left some comments.

File: functions/typescript/modules/module2/index.complete.ts (hunk 1 above)
Comment (dreamorosi) [reply functions/typescript/modules/module2/index.complete.ts:33]: Yes, good catch - I need to fix it also in the workshop text.

**Comment by leandrodamascena** [comment]
Hello @dreamorosi! Can you merge it please?

I'll probably open a new one on Friday, but I'm good so far.

**Comment by dreamorosi** [comment]
Thank you for working on this Leo!

**Pull request merged**